### DISCUSSION TITLE
[archived] "Argo Server SSO with Argo CD Dex" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#13062] [General]
# en/stable/argo-server-sso-argocd/

https://argo-workflows.readthedocs.io/en/stable/argo-server-sso-argocd/

**Comment by jiantang99** [comment]
Id field in argocd-cm should be clientId given by OIDC, the same one used in the secret.

**Reply by agilgur5** [reply]
Yes this was clarified in https://github.com/argoproj/argo-workflows/pull/12843, which is visible in the `latest` version of the docs: https://argo-workflows.readthedocs.io/en/latest/argo-server-sso-argocd/

**Reply by crdsvc** [reply]
@jiantang - Im using github sso for argo cd - but for configuring workflows, do I need to provide that client_id ?

**Reply by agilgur5** [reply]
You would provide the `staticClient` ID, as the docs say.
In this scenario, Workflows is a client of Dex which is a client of GitHub.

**Comment by jiantang99** [comment]
Is there an api to call to obtain the token programatically with sso installed with dex in argo workflow? (I.e., Bearer v2.....)

**Reply by agilgur5** [reply]
There can't be an API for that since that would require authentication which would require a token -- chicken-and-egg problem.

If you're looking for a [client token](https://argo-workflows.readthedocs.io/en/latest/access-token/), which is unrelated to SSO, you can programmatically create and retrieve it via `kubectl`, per the directions.
